lib/spack/spack/patch.py

```diff
@@ -90,7 +91,8 @@ class FilePatch(Patch):
     def __init__(self, pkg, path_or_url, level):
         super(FilePatch, self).__init__(pkg, path_or_url, level)
 
-        pkg_dir = spack.repo.dirname_for_package_name(pkg.name)
+        m = inspect.getmodule(pkg)
```

Comment:
> Yes, there was even a comment above:

Yep. I missed that initially. I prefer the initial summary to be updated rather than applying all deltas in the comments (although I definitely should have been more careful).

> So: should I move this to another PR and fix spack.repo.dirname_for_package to take a Package instead of a string?

That is my preference